sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleAssignment.java

```diff
@@ -24,14 +23,12 @@
      * @param type The type of this {@link KeyVaultRoleAssignment role assignment}.
      * @param properties {@link KeyVaultRoleAssignmentProperties properties} of this {@link KeyVaultRoleAssignment
      * role assignment}.
-     * @param roleScope The {@link KeyVaultRoleScope scope} of this {@link KeyVaultRoleAssignment role assignment}.
      */
-    public KeyVaultRoleAssignment(String id, String name, String type, KeyVaultRoleAssignmentProperties properties, KeyVaultRoleScope roleScope) {
+    public KeyVaultRoleAssignment(String id, String name, String type, KeyVaultRoleAssignmentProperties properties) {
```

Comment:
Why is this publicly creatable? The only way they should be able to create one is via `KeyVaultAccessControlClient.createRoleAssignment`.

Reply:
All these return types in Java have to be public so that deserialization can work properly. Even if users can create them they are not accepted as input parameters anywhere.

Reply:
Now that I think of it, given the way we create these types, the constructors can be package private.

Reply:
Actually, in this case it's because the model is in a different package than the client, which means we need to make the constructor and the class `public`.